lib/appium/AppiumRunner.js

```diff
@@ -0,0 +1,491 @@
+        // needed directive is not there and there is no default-src directive
+        // but the CSP tag is till present
+        // just adding needed directive to a start of CSP tag content
+        cspContent = cspContent.replace(cspTagOpening, cspTagOpening + directive + " " + source + "; ");
+        fs.writeFileSync(cspFile, cspContent, util.DEFAULT_ENCODING);
+    } else {
+        // no CSP tag, skipping
+        logger.normal("paramedic-appium: WARNING: No CSP tag found.");
+    }
+}
+
+function setPreference(appPath, preference, value) {
+    var configFile = getConfigPath(appPath);
+    var xml = parseElementtreeSync(configFile);
+    var pref = xml.find("preference[@name=\"" + preference + "\"]");
+
+    logger.normal("paramedic-appium: Setting \"" + preference + "\" preference to \"" + value + "\"");
+
+    if (!pref) {
+        pref = new elementTree.Element("preference");
+        pref.attrib.name = preference;
+        xml.getroot().append(pref);
+    }
+    pref.attrib.value = value;
+
+    // write the changes
+    fs.writeFileSync(configFile, xml.write({indent: 4}), util.DEFAULT_ENCODING);
+}
+
+function permitAccess(appPath, origin) {
+    var configFile = getConfigPath(appPath);
+    var xml = parseElementtreeSync(configFile);
+    var rule = xml.find("access[@origin=\"" + origin + "\"]");
+
+    logger.normal("paramedic-appium: Adding a whitelist 'access' rule for origin: " + origin);
+
+    if (rule) {
+        logger.normal("paramedic-appium: It is already in place");
+    } else {
+        rule = new elementTree.Element("access");
+        rule.attrib.origin = origin;
+        xml.getroot().append(rule);
+        fs.writeFileSync(configFile, xml.write({indent: 4}), util.DEFAULT_ENCODING);
+    }
+}
+
+function runCommand(command, appPath) {
+    if (appPath) {
+        shell.pushd(appPath);
+    }
+    shell.exec(command);
+    if (appPath) {
+        shell.popd();
+    }
+}
+
+function isFailFastError(error) {
+    if (error && error.message) {
+        return error.message.indexOf("Could not find a connected") > -1 ||
+            error.message.indexOf("Bad app") > -1;
+    }
+    return false;
+}
+
+function killProcess(procObj, killSignal, callback) {
+    if (procObj && procObj.alive) {
+        procObj.alive = false;
+        setTimeout(function () {
+            kill(procObj.process.pid, killSignal, callback);
+        }, 1000);
+    } else {
+        callback();
+    }
+}
+
+function installAppiumServer(success, fail) {
+    logger.normal("paramedic-appium: Installing Appium server...");
+    shell.pushd(path.join(__dirname, "../.."));
+    shell.exec("npm install appium", { async: true, silent: !global.VERBOSE }, function (code, stdout) {
+        if (code !== 0) {
+            fail("Couldn't install Appium server:\n" + stdout);
+        } else {
+            success();
+        }
+    });
+    shell.popd();
+}
+
+function addCallback(reporter, callback) {
```

Comment:
It might be better to place this function into 'Reporters.js' (a static method?)